contracts/StakingToken.sol

```diff
@@ -94,7 +98,7 @@ contract StakingToken is ERC20Upgradeable {
             requestedDuration = 14;
         }
         uint256 reward = calculateReward(_stake, requestedDuration);
-        ERC20Upgradeable.transferFrom(owner, _msgSender(), reward); // transferring Allocate tokens
+        _mint(_msgSender(), reward); // transferring Allocate tokens
```

Comment:
We need to remove the minting of the full supply to the owner on line 44, no?

Reply:
Well, it is standard that the address that deploys the contract, is the owner of the initially minted tokens. 
https://docs.openzeppelin.com/contracts/4.x/erc20

Perhaps we could mint the tokens to the contract address. Is this what you are thinking? I still like using the mint function in this manner as we will never have to worry about reaching our max on total supply.

Reply:
But we are minting the total supply when deploying, doesn’t that mean that the following mints from staking will exceed the total supply?

Reply:
Ah I think I get, totalSupply means “initial supply” not “max supply” right?

Reply:
Thats right. There is no cap